### PR TITLE
LocalizationResourceManager #93

### LINKED ISSUE #93: [Proposal] LocalizationResourceManager
# LocalizationResourceManager

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
  * [ ] iOS Support
  * [ ] Android Support
  * [ ] macOS Support
  * [ ] Windows Support
* [ ] Unit Tests: Not Started
* [ ] Sample: Not Started
* [ ] Documentation: Not Started

## Summary
[summary]: #summary

The LocalizationResourceManager class is a helper class that enables users to respond to culture changes at runtime. This class is typically used by the TranslateExtension class and LocalizedString.

## Detailed Design
[design]: #detailed-design

### LocalizationResourceManager.shared.cs

```cs
public class LocalizationResourceManager : ObservableObject
{
  public static LocalizationResourceManager Current;

  private LocalizationResourceManager();

  public void Init(ResourceManager resource);
  public void Init(ResourceManager resource, CultureInfo initialCulture);
  public string GetValue(string text);

  public string this[string text] => GetValue(text);

  public CultureInfo CurrentCulture { get; set; }
}
```

## Usage Syntax
[usage]: #usage-syntax

### XAML Usage

N/A

### C# Usage

```cs
LocalizationResourceManager.Current.PropertyChanged += (_, _) => AppResources.Culture = LocalizationResourceManager.Current.CurrentCulture;
LocalizationResourceManager.Current.Init(AppResources.ResourceManager);

LocalizationResourceManager.Current.CurrentCulture = newCulture;
```

## Drawbacks
[drawbacks]: #drawbacks

This feature seems to duplicate the built-in localization tools with RESX.

It also implements `ObservableObject` which will not be included in `CommunityToolkit.Maui` because it is already implemented in `CommunityToolkit.Mvvm`.

## Alternatives
[alternatives]: #alternatives

**RESX**
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/localization/text?pivots=macos?WT.mc_id=mobile-0000-bramin

```cs
CultureInfo.DefaultThreadCurrentCulture = "en-us";
CultureInfo.DefaultThreadCurrentUICulture = "en-us";
```

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**
### Description of Change ###

Adds LocalizationResourceManager

 ### Linked Issues ###
 

 - Fixes CommunityToolkit/dotnet#312

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

**Review by pictos — CHANGES_REQUESTED** [review]
@VladislavAntonyuk thanks for moving this forward, but if I'm not wrong we discussed the idea of moving this to the Common CommunityToolkit, since this feature can be used across the .NET ecosystem and adds more value. I would like to hear what you think and also the other team members.

**Comment by brminnick** [comment]
Hey Vlad! This feature hasn't been approved yet. 

I'm adding the `do not merge` label to this until the Proposal has been voted on and approved.

**Comment by brminnick** [comment]
Closing this PR as it is being moved to `CommunityToolkit.Common`: https://github.com/CommunityToolkit/dotnet/issues/312

**Pull request closed without merging**
